fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryPlanWithComparisons.java

```diff
@@ -21,19 +21,38 @@
 package com.apple.foundationdb.record.query.plan.plans;
 
 import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.query.expressions.Comparisons;
 import com.apple.foundationdb.record.query.plan.ScanComparisons;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
 
 import javax.annotation.Nonnull;
+import java.util.Set;
 
 /**
- * A query plan that uses {@link ScanComparisons} to drive some scan of the record store.
+ * A query plan that uses {@link ScanComparisons} or {@link Comparisons.Comparison} to drive some scan of
+ * the record store.
  */
 @API(API.Status.EXPERIMENTAL)
 public interface RecordQueryPlanWithComparisons extends RecordQueryPlan {
-    @Nonnull
-    ScanComparisons getComparisons();
 
     default boolean hasComparisons() {
+        return hasScanComparisons();
+    }
+
+    default Set<Comparisons.Comparison> getComparisons() {
```

Comment:
This method has a rather strange contract. The main issue is that comparisons are associated with a particular position. For equality, their position in the list, for inequality, the number of equalities that proceed. And this just flattens that.

As used by `computeSargedInBindings` and `compareInOperator`, only certain equality comparisons are looked at and their parameter names picked out.

It seems that we need to know for a given plan how it uses any parameters that it depends on.
* Filter plan evaluates expression including the parameter.
* Index plan binds part of the scan range to the parameter.
* Lucene plan builds the search query to include the parameter's value.

An (imagined) prefix string index scan would have both the first and second. In this case, I believe we consider the index part enough to justify its use.

Reply:
I view this method as the `getCorrelatedTo()` of sorts in the old planner realm. The old `getComparisons()` has been renamed to `getScanComparisons()`. So the old API is still intact.
There are `Comparisons` inside `ScanComparisons` (inequalities) that are a set already.
I would like to keep the comparisons in a set here and then filter them later on by use case rather than only filtering them here and then have to open up this collector for other use cases in the future.

Reply:
Maybe it's harmless noise, but compared to `getCorrelatedTo` concretely, `getComparisons` is full of a bunch of comparisons to literal values. And, to me, the set cardinality difference when you happen to have two `= 1` versus `= 1` and `= 2` or `= 1` and `>= 1` suggests that it isn't really meaningful. I suppose the idea put forward is that this is just part of some API and every particular use case ends up wrapping the result in what it takes as important before doing any reasoning.